src/addon-manager.js

```diff
@@ -760,12 +760,14 @@ class AddonManager extends EventEmitter {
       this.adapters.delete(a.id);
     }
 
-    const plugin = this.getPlugin(packageName);
-    if (plugin) {
-      plugin.kill();
-    }
+    const cleanup = () => {
+      const plugin = this.getPlugin(packageName);
+      if (plugin) {
+        plugin.kill();
```

Comment:
I think we should also log something here, like plugin didn't shutdown cleanly, killing